holdings_maintenance/pds4/pds4linkshelf.py

```
@@ -0,0 +1,1270 @@
+TARGET_REGEX2 = re.compile(r'^ *,? *' + PATTERN, re.I)
+
+# Match pattern for one or more file names embedded in a row of a text file.
+# A file name begins with a letter, followed by any number of letters, digits,
+# underscore or dash. Unless the name is "Makefile", it must have one or more
+# extensions, each containing one or more characters. It can also have any
+# number of directory prefixes separate by slashes.
+
+LINK_REGEX = re.compile(r'(?:|.*?[^/@\w\.])/?(?:\.\./)*(([A-Z0-9][-\w]+/)*' +
+                        r'(makefile\.?|[A-Z0-9][\w-]*(\.[\w-]+)+))', re.I)
+
+EXTS_WO_LABELS = set(['.XML', '.CAT', '.FMT', '.SFD'])
```

Ruff (0.14.5)

Unnecessary list literal (rewrite as a set literal)

Rewrite as a set literal

(C405)